contractsV5/FeeHandler.sol

```diff
@@ -0,0 +1,189 @@
+import "./IERC20.sol";
+
+contract FeeHandler is IFeeHandler, Utils {
+
+    IKyberDAO public kyberDAOContract;
+    // if I use the interface, I am not able to get the KyberNetwork internal contract from the proxy..
+    KyberNetworkProxy public kyberNetworkProxyContract;
+    address public knc;
+    address payable public burnAddress;
+    uint public burnBlockInterval;
+    uint public lastBurnBlock;
+    uint constant ETH_TO_BURN = 10**19;
+
+    uint public brrAndEpochData;
+    uint constant BITS_PER_PARAM = 64;
+
+    uint public totalRebates;
+    mapping(address => uint) public totalRebatesPerRebateWallet;
+    uint public totalRewards;
+    mapping(uint => uint) public totalRewardsPerEpoch;
+
+
+    constructor(
+        IKyberDAO _kyberDAOContract,
+        KyberNetworkProxy _kyberNetworkProxyContract,
+        address _knc,
+        address payable _burnAddress,
+        uint _burnBlockInterval
+    ) public
+    {
+        kyberDAOContract = _kyberDAOContract;
+        kyberNetworkProxyContract = _kyberNetworkProxyContract;
+        knc = _knc;
+        burnAddress = _burnAddress;
+        burnBlockInterval = _burnBlockInterval;
+        lastBurnBlock = block.number;
+    }
+
+    modifier onlyDAO {
+        require(
+            msg.sender == address(kyberDAOContract),
+            "Only the DAO can call this function."
+        );
+        _;
+    }
+
+    modifier onlyKyberNetwork {
+        require(
+            msg.sender == address(kyberNetworkProxyContract.kyberNetworkContract()),
+            "Only the internal KyberNetwork contract can call this function."
+        );
+        _;
+    }
+
+
+    function encodeData(uint _burn, uint _reward, uint _epoch, uint _expiryBlock) public pure returns (uint) {
+        return (((((_burn << BITS_PER_PARAM) + _reward) << BITS_PER_PARAM) + _epoch) << BITS_PER_PARAM) + _expiryBlock;
+    }
+
+    function decodeData() public view returns(uint, uint, uint, uint) {
+        uint expiryBlock = brrAndEpochData & (1 << BITS_PER_PARAM) - 1;
+        uint epoch = (brrAndEpochData / (1 << BITS_PER_PARAM)) & (1 << BITS_PER_PARAM) - 1;
+        uint rewardInBPS = (brrAndEpochData / (1 << BITS_PER_PARAM << BITS_PER_PARAM)) & (1 << BITS_PER_PARAM) - 1;
+        uint burnInBPS = (brrAndEpochData / (1 << BITS_PER_PARAM << BITS_PER_PARAM << BITS_PER_PARAM)) & (1 << BITS_PER_PARAM) - 1;
+        return (burnInBPS, rewardInBPS, epoch, expiryBlock);
+    }
+
+    event AccumulateReserveRebate(address rebateWallet, uint rebateAmountWei);
+
+    // Todo: future optimisation to accumulate rebates for 2 rebate wallet
+    // encode totals, 128 bits per reward / rebate
+    function handleFees(address[] calldata eligibleWallets, uint[] calldata rebatePercentages) external payable onlyKyberNetwork returns(bool) {
+        // Decoding BRR data
+        (uint burnInBPS, uint rewardInBPS, uint epoch, uint expiryBlock) = decodeData();
+        uint rebateInBPS = BPS - rewardInBPS - burnInBPS;
```

Comment:
I thought we decided burn will be calculated.

Reply:
It's the same either way... I can change it to burn if you want. Just need to change the encode decode function

Reply:
As discussed in our call, I will change this to calculate burn